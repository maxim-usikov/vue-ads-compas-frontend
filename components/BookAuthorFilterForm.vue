<template>
  <VForm ref="form" v-model="valid">
    <VTextField v-model="title" label="Book title" required />

    <VTextField v-model="authorName" label="Author name" required />

    <VBtn color="primary" :disabled="!valid" @click="handleFilter">Filter</VBtn>
    <VBtn color="secondary" @click="handleReset">Reset</VBtn>
  </VForm>
</template>

<script>
export default {
  name: 'BookAuthorFilterForm',

  data() {
    return {
      valid: false,
      title: this.$route.query.title || '',
      authorName: this.$route.query.authorName || ''
    }
  },

  methods: {
    handleFilter() {
      const { title, authorName } = this

      this.$emit('filter', { title, authorName })
    },

    handleReset() {
      const title = undefined
      const authorName = undefined

      this.$refs.form.reset()
      this.$emit('filter', { title, authorName })
    }
  }
}
</script>
