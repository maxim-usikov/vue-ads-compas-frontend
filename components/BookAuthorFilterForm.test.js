import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'

import BookAuthorFilterForm from './BookAuthorFilterForm.vue'

const localVue = createLocalVue()
localVue.use(VueRouter)

describe('BookAuthorFilterForm', () => {
  let $c

  beforeEach(() => {
    const router = new VueRouter({})
    $c = shallowMount(BookAuthorFilterForm, { localVue, router })
  })

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy()
  })

  it('name is `BookAuthorFilterForm`', () => {
    expect($c.name()).toBe('BookAuthorFilterForm')
  })

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot()
  })
})
