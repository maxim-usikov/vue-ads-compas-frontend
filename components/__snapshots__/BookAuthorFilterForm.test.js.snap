// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookAuthorFilterForm renders correctly 1`] = `
<vform-stub>
  <vtextfield-stub
    label="Book title"
    required=""
    value=""
  />
   
  <vtextfield-stub
    label="Author name"
    required=""
    value=""
  />
   
  <vbtn-stub
    activeclass="v-btn--active"
    color="primary"
    disabled="true"
    tag="button"
    type="button"
  >
    Filter
  </vbtn-stub>
   
  <vbtn-stub
    activeclass="v-btn--active"
    color="secondary"
    tag="button"
    type="button"
  >
    Reset
  </vbtn-stub>
</vform-stub>
`;
